services/dask-sidecar/tests/unit/test_file_utils.py

```diff
@@ -335,7 +345,33 @@ async def test_pull_compressed_zip_file_from_remote(
     assert download_folder.exists()
     dst_path = download_folder / faker.file_name()
     await pull_file_from_remote(
-        src_url, dst_path, mocked_log_publishing_cb, remote_parameters.s3_settings
+        src_url=src_url,
+        target_mime_type=None,
+        dst_path=dst_path,
+        log_publishing_cb=mocked_log_publishing_cb,
+        s3_settings=remote_parameters.s3_settings,
+    )
+    assert not dst_path.exists()
+    for file in download_folder.glob("*"):
+        assert file.exists()
+        assert file.name in file_names_within_zip_file
+    mocked_log_publishing_cb.assert_called()
+    mocked_log_publishing_cb.reset_mock()
+
+    # USE-CASE 3: if destination is a zip, but we pass a target mime type that is not, then we decompress
```

Comment:
Is this description correct?
I would expect not to decompress.

Reply:
it is correct. it uses the target mime type, which is what the service expects. Example:
my service has an input port that is defined as in:
```type: data:*/*```
Then it means it wants to have it unzipped.

If the service is defined as:
```type: data:application/zip```
then it means it wants a zip file